### ISSUE TITLE
Multiline function definition headers

### ISSUE THREAD

**Issue by RubenVanEldik** [astral-sh/ruff#13149]
I have a question about a quite opinionated formatting issue. Apologies if this has been discussed before, but it's quite hard to search for it in the exiting issues.

When a list (or, set, tuple, etc) gets too long for a single line, it becomes a multiline statement where each item is on its own line. However, when the function arguments get too long (which happens very often when you're adding types), the function arguments get all moved to their own line, if possible.

I think this intermediate solution is the worst of both worlds because the number of lines are tripled but the line length is often barely shorter. Which makes the whole function harder to read.

You can fix this yourself already by adding a comma after the last argument, but I would like to propose that if a function definition header does not fit on a single line, each argument should get it's own line. Similar to lists, etc.

Before formatting:
```py
my_list_variable = ["item_one", "item_two", "item_three", "item_four", "item_five", "item_six"]

def my_func(arg_1: int, arg_2: int, arg_3: int, arg_4: int, arg_5: int, arg_6: int, arg_7: int):
    pass
```

After formatting:
```py
my_list_variable = [
    "item_one",
    "item_two",
    "item_three",
    "item_four",
    "item_five",
    "item_six",
]


def my_func(
    arg_1: int, arg_2: int, arg_3: int, arg_4: int, arg_5: int, arg_6: int, arg_7: int
):
    pass
```

My proposed fromatting:
```py
my_list_variable = [
    "item_one",
    "item_two",
    "item_three",
    "item_four",
    "item_five",
    "item_six",
]


def my_func(
    arg_1: int,
    arg_2: int,
    arg_3: int,
    arg_4: int,
    arg_5: int,
    arg_6: int,
    arg_7: int,
):
    pass

```

([Ruff playground link](https://play.ruff.rs/94e8d3d7-a048-409e-a321-f0ac24a9db86))

* List of keywords you searched for before creating this issue. `function definition header', 'function signature'

**Comment by AlexWaygood** [comment]
Hi, thanks for opening the issue!

As with many formatting decisions, this is unfortunately quite subjective. Many people prefer the "intermediate" solution rather than their function definition immediately being exploded across multiple lines when it goes a single character over the line length. In this style decision -- as with most of our style decisions currently -- we follow what Black does.

However, are you aware of the formatter's behaviour around trailing commas? If you have a trailing comma at the end of your parameter list, the formatter will always split the parameter list over multiple lines, even if it's well below the maximum line length you've specified: https://play.ruff.rs/27f31e3e-5bf5-40e2-b142-4b885563940f

**Comment by RubenVanEldik** [comment]
Hi Alex, thank you for your response!

It is indeed very subjective! I wanted to see if this could be a discussion for Ruff, but it makes total sense to follow Black in this regard. So I'll close the issue!

For now I'll just instruct all my colleagues manually to add the comma whenever it doesn't fit on a single line anymore :)

**Comment by AlexWaygood** [comment]
> For now I'll just instruct all my colleagues manually to add the comma whenever it doesn't fit on a single line anymore :)

Ah, you can enforce it with one of our linter rules ;) https://docs.astral.sh/ruff/rules/missing-trailing-comma/

**Comment by RubenVanEldik** [comment]
You're completely right.. Somehow I have added this rule to our ignore list a few months ago.

Sorry for discussing an a rule that I actively ignored myself... 🙄🙄

**Comment by AlexWaygood** [comment]
hehe, no worries!
